aix.js

```diff
@@ -0,0 +1,78 @@
+"use strict";
+
+const getSqlStatement = family => {
+  let sql =
+    "select NEXT_HOP from QSYS2.NETSTAT_ROUTE_INFO where ROUTE_TYPE='DFTROUTE'";
+
+  if (family === "v4") {
+    sql += " and CONNECTION_TYPE='IPV4'";
+  } else {
+    sql += " and CONNECTION_TYPE='IPV6'";
+  }
+
+  return sql;
+};
+
+const getGatewayInformationAsync = async family => {
+  return new Promise((resolve, reject) => {
+    const idbConnector = require("idb-connector");
+    const dbconn = new idbConnector.dbconn();
+    dbconn.conn("*LOCAL");
+
+    const sql = getSqlStatement(family);
+
+    const stmt = new idbConnector.dbstmt(dbconn);
+
+    stmt.exec(sql, async results => {
+      stmt.close();
+      dbconn.disconn();
+      dbconn.close();
+
+      if (results && results[0] && results[0].NEXT_HOP) {
+        resolve({
+          gateway: results[0].NEXT_HOP
```

Comment:
Documentation says that the special value `*DIRECT` is "the next hop value of a route that is automatically created.". Maybe this special value should be handled similar to the [Linux](https://github.com/silverwind/default-gateway/blob/9d3c3a5dd46bbc09f93bce68f35cace7a7f06593/linux.js#L22) case of `default via dev eth1`? I realize that returning the matching interface address is not totally correct (practically, there is no default gateway address an route via interface), but it's better than failing, I suppose.

Alternatively, we could just fail of `*DIRECT` is encountered.

Reply:
As far as I‘ve seen in the database, there is an entry with value `*DIRECT`, that‘s why I filter on `ROUTE_TYPE=‘DFTROUTE‘`

Reply:
Maybe add `!='*DIRECT'` to the query as a safeguard, if it is necessary.

Reply:
Added the `NEXT_HOP` filter